plasmapy/atomic/classes.py

```diff
@@ -0,0 +1,250 @@
+    InvalidIonError,
+    ChargeError,
+    MissingAtomicDataError,
+)
+
+from .particles import (
+    _Particles,
+    _special_particles,
+    _leptons,
+    _antileptons,
+    _baryons,
+    _antibaryons,
+    _particles,
+    _antiparticles,
+    _fermions,
+    _bosons,
+    _neutrinos,
+    _antineutrinos,
+)
+
+from .elements import (_Elements, _atomic_symbols, _atomic_symbols_dict)
+from .isotopes import _Isotopes
+
+from .parsing import (
+    _dealias_particle_aliases,
+    _parse_and_check_atomic_input,
+)
+
+
+class Particle():
+    r"""A class for individual particles or antiparticles."""
+
+    def __init__(self,
+                 argument: Union[str, int],
+                 mass_numb: int = None,
+                 Z: int = None):
+
+        self._original_argument = argument
+        self._original_mass_number = mass_numb
+        self._original_integer_charge = Z
+        self._particle_symbol = None
+        self._atomic_symbol = None
+        self._isotope_symbol = None
+        self._ion_symbol = None
+        self._unicode_symbol = None
+        self._element_name = None
+        self._atomic_number = None
+        self._mass_number = None
+        self._lepton_number = None
+        self._baryon_number = None
+        self._is_stable = None
+        self._is_antimatter = None
+        self._integer_charge = None
+        self._electric_charge = None
+        self._mass = None
+        self._half_life = None
+        self._lepton = None
+        self._antilepton = None
+        self._baryon = None
+        self._antibaryon = None
+        self._fermion = None
+        self._generation = None
+        self._periodic_table_group = None
+        self._periodic_table_period = None
+
+        particle = _dealias_particle_aliases(argument)
+
+        is_special_particle = particle in _special_particles
+
+        if is_special_particle and (mass_numb is not None or Z is not None):
+            raise InvalidParticleError()
+
+        if is_special_particle:
+            self._particle_symbol = particle
+            self._name = _Particles[particle]['name']
+            self._spin = _Particles[particle]['spin']
+            self._class = _Particles[particle]['class']
+            self._lepton_number = _Particles[particle]['lepton number']
+            self._baryon_number = _Particles[particle]['baryon number']
+            self._integer_charge = _Particles[particle]['charge']
+            self._mass = _Particles[particle]['mass']
+
+            self._is_stable = _Particles[particle]['is_stable']
+
+            self._half_life = _Particles[particle]['half-life']
+            self._is_antimatter = _Particles[particle]['antimatter']
+
+            if particle in _leptons + _antileptons:
+                self._is_lepton = True
+                self._generation = _Particles[particle]['generation']
+            else:
+                self._is_lepton = False
+
+            if particle in _fermions:
+                self._is_fermion = True
```

Comment:
`self._is_fermion = particle in _fermions`, without the if here? The previous is understandable as you also need to allocate `self._generation`.

Reply:
Good catch!  The previous can still be simplified/shortened using this strategy.

I'm still unlearning Fortran, it would seem!